app/services/submission/trait_score_parser.rb

```diff
@@ -82,4 +149,22 @@ def csv
   def input
     @input ||= File.open(@upload.file.path)
   end
+
+  def detect_replication(header_columns)
+    header_columns.any?{ |column_name| column_name && column_name.index(/rep\d+$/) }
+  end
+
+  def split_to_trait_and_replicate(column_name)
+    replicate_number = 0
+    return ['',0] if column_name.blank?
+    trait_name = column_name.strip.gsub(/rep\d+$/) do |replicate|
```

Comment:
I actually like my use of gsub with block better - it is actually simpler in my opinion compared to your version.

However, I did change placeholder values generator - before I opted for no-space, all-underscore placeholders so it is easier to click-select it in a tool like Excel and replace. But I changed my mind in the middle of the process, as you see - now I removed underscores completely. I think the template is easier to comprehend this way.